flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTest.java

```diff
@@ -846,6 +817,23 @@ public void testSetsUserCodeClassLoaderForTimerThreadFactory() throws Throwable
 	//  Test Utilities
 	// ------------------------------------------------------------------------
 
+	private static void waitUntilExecutionState(Task task, ExecutionState exceptedState, Deadline deadline) {
+		while (deadline.hasTimeLeft()) {
+			if (exceptedState == task.getExecutionState()) {
+				return;
+			}
+
+			try {
+				Thread.sleep(Math.min(deadline.timeLeft().toMillis(), 200));
```

Comment:
Just a little reminder, using Thread.sleep() in Junit Test thread sometimes may leads to some strange behavior. I am not sure if it is safe here.

Reply:
Thanks! I change waiting style to notify style, it would be more stable.

Reply:
@tillrohrmann yes it has been addressed, please see the following commit.